src/i18n/i18nConfig.ts

```diff
@@ -1,13 +1,39 @@
 import i18n from "i18next";
 import LanguageDetector from "i18next-browser-languagedetector";
+import type { CustomDetector } from "i18next-browser-languagedetector";
 import { initReactI18next } from "react-i18next";
+import api from "~/common/lib/api";
 
 import en_common from "./locales/en/common.json";
 import hi_common from "./locales/en/common.json";
 import en_components from "./locales/en/components.json";
 import en from "./locales/en/translation.json";
 import hi from "./locales/hi/translation.json";
 
+export const settings: CustomDetector = {
+  name: "settings",
+
+  lookup(_options) {
+    let settingsLang;
+    api.getSettings().then((setting) => {
+      i18n.changeLanguage(setting.locale);
+      settingsLang = setting.locale;
+    });
+    return settingsLang;
```

Comment:
I am pretty sure this does not work. did you test it?

api.getSettings() is called asynchronously. this means the return here happens before api.getSettings() finished. 
settingsLang can not be available here.

Reply:
Yes you are right again, I tested this and `settingsLang` is undefined when it gets returned. But the functionality of this PR still works, it was working when we were returning `undefined`... should we say that it is OK because it achieves the goal of this PR which is to allow users to select the language settings via the Alby Settings page? Or should we debug further to try and understand this CustomDetector class better?

Reply:
from how I understand it accidentally works because of a side-effect and not because of having this Detector. 

it only works because the lookup calls 
```
api.getSettings().then((setting) => {
  i18n.changeLanguage(setting.locale);
});
```

if that's how we change the language the we do not need to do that in such a detector. 
So I don't think this PR can be merged like this.

Somebody else looking at this code in a few months will also not understand this and will be confused.

Reply:
Yes that is how we change the language when the user selects one in the Settings, but the problem is - will that persist on refresh? I thought the reason we needed this custom detector is that i18n will not detect the Alby user settings when they close the browser or on refresh? I can test it without this custom detector and see if it still works though.

Reply:
as you said your custom detector does not work. lookup does return `undefined` so the next detector will be used etc. 

it only works because this custom detector executes some code as a side effect. it executes code to load the language form the settings and calls i18n.changeLanguage. 
but the detector is the wrong place to trigger that.

Reply:
Ok I think I have removed all the not needed code now, I tested it and it looks like it works - you can select a language in the Alby settings and then refresh and the language you selected will persist.

I guess we over-thought this from the beginning, we didn't need all this fancy custom detector class stuff in there...

Please have a look and let me know what you think now, thanks!

Reply:
actually that looks good to me! good job! 

@pavanjoshi914 can you review this and confirm?